presidio-analyzer/presidio_analyzer/pattern_recognizer.py

```diff
@@ -110,21 +110,23 @@ def build_regex_explanation(
                                           validation_result=validation_result)
         return explanation
 
-    def __analyze_patterns(self, text):
+    def __analyze_patterns(self, text, flags=None):
         """
         Evaluates all patterns in the provided text, including words in
          the provided blacklist
 
         :param text: text to analyze
+        :param flags: regex flags
         :return: A list of RecognizerResult
         """
+        flags = flags if flags else re.DOTALL | re.MULTILINE
         results = []
         for pattern in self.patterns:
             match_start_time = datetime.datetime.now()
             matches = re.finditer(
                 pattern.regex,
                 text,
-                flags=re.IGNORECASE | re.DOTALL | re.MULTILINE)
+                flags=flags)
```

Comment:
Why remove the IGNORECASE flag?

Reply:
It prevents one from making a case sensitive regex. If you want a case insensitive regex you can add the modifier "(?i)", so you can now do either case sensitive or case insensitive whereas before you could only do case sensitive.

Reply:
Case insensitive regex would theoretically improve recall (and might reduce precision). To me it sounds like it should be the default. I'll consult with the team. Do you think each regex should explicitly add (?i) to become case insensitive?

Reply:
yes, I think if you want a case-insensitive regex then you should write the regex to be case insensitive.  This was actually a pain point for me, because it was buried deep within the code, so I was trying to write a case-sensitive regex and the case sensitivity was automatically ignored.